syntax = "proto3";

package jiaa.text;

option java_multiple_files = true;
option java_package = "com.jiaa.text";

// TextAIService handles text-based AI operations
service TextAIService {
  // Break down a complex goal into smaller sub-tasks
  rpc GenerateSubgoals (GoalRequest) returns (GoalResponse);
  
  // Direct Chat with Persona (Text-to-Text)
  rpc Chat (ChatRequest) returns (ChatResponse);

  // Generate Technical Quiz
  rpc GenerateQuiz (QuizRequest) returns (QuizResponse);
}

message GoalRequest {
  string goal_text = 1; // e.g. "Make a todo app"
}

message GoalResponse {
  repeated string subgoals = 1; // e.g. ["Setup React", "Configure Node.js"]
}

message ChatRequest {
  string text = 1;      // e.g. "주인님이 핸드폰을 보고 계십니다"
  string client_id = 2; // Optional
}

message ChatResponse {
  string message = 1;          // AI Response text
  string intent = 2;           // COMMAND, CHAT...
  string action_code = 3;      // OPEN_APP, KILL_APP...
  string action_detail = 4;    // Target
  string emotion = 5;          // Emotion Tag
  string judgment = 6;         // STUDY, PLAY, NEUTRAL
}

message QuizRequest {
  string topic = 1;       // e.g., "Python Asyncio"
  string difficulty = 2;  // "Easy", "Medium", "Hard"
}

message QuizItem {
  string question = 1;
  repeated string options = 2;
  string answer = 3; 
  string explanation = 4;
}

message SubgoalQuiz {
  string subgoal = 1;
  repeated QuizItem quizzes = 2;
}

message QuizResponse {
  repeated SubgoalQuiz items = 1;
}
